### ISSUE TITLE
Support logic NOT operator

### ISSUE THREAD

**Issue by liouxiao** [hyperjumptech/grule-rule-engine#94]
I think it should be easy to support logic NOT (!) operator.

The rule would looks like:

```
rule RuleOne "Some rule description." salience 10 {
    when
        !(someContext.attributeA.Match("123") &&
          someContext.attributeA.Contains("1"))
    then
        ...
```

Without NOT operator, I have to write the rule like:
```
rule RuleOne "Some rule description." salience 10 {
    when
        someContext.attributeA.Match("123")==false ||
        someContext.attributeA.Contains("1")==false
    then
        ...
```

**Comment by hellocowboy** [comment]
there i find a better engine to fit you!
https://github.com/rencalo770/gengine
document: https://rencalo770.github.io/gengine_en

**Comment by liouxiao** [comment]
> there i find a better engine to fit you!
> https://github.com/rencalo770/gengine
> document: https://rencalo770.github.io/gengine_en

I know "gengine" - it obviously borrowed some design (or initial implementation) here. 

But that's not the key issue of gengine - it even has no license statement - there's somehow risk to use it.

**Comment by hellocowboy** [comment]
first ,I am just a user!
i think, you may need to&nbsp; read the doc in detail, and compare two sides implements.&nbsp;
it has&nbsp;license statement, its license is BSD.&nbsp;
All above is just a suggestion, not must be.let us stop arguments and just choose ourself needed.

thanks for your reply!&nbsp; 








------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Xiao Xi LIU"<notifications@github.com&gt;; 
发送时间: 2020年11月4日(星期三) 下午3:02
收件人: "hyperjumptech/grule-rule-engine"<grule-rule-engine@noreply.github.com&gt;; 
抄送: "Clivia"<956517404@qq.com&gt;; "Comment"<comment@noreply.github.com&gt;; 
主题: Re: [hyperjumptech/grule-rule-engine] Support logic NOT operator (#94)





  
there i find a better engine to fit you!
 https://github.com/rencalo770/gengine
 document: https://rencalo770.github.io/gengine_en
  
I know "gengine" - it obviously borrowed some design (or initial implementation) here.
 
But that's not the key issue of gengine - it even has no license statement - there's somehow risk to use it.

**Comment by liouxiao** [comment]
> first ,I am just a user! i think, you may need to&nbsp; read the doc in detail, and compare two sides implements.&nbsp; it has&nbsp;license statement, its license is BSD.&nbsp; All above is just a suggestion, not must be.let us stop arguments and just choose ourself needed. thanks for your reply!&nbsp;

No offense, I'm sorry if my reply made you unhappy - I'd not argue which one is better or not - everyone has the right to make choices.

I DID search in the project and found there's only a tiny BSD icon displayed in the README page, I think it should attach a LICENSE file at least. Anyway, thank you for let me know it.

Except the license issue, I also think gengine is a good project - the same as you.

B.R.

**Comment by hellocowboy** [comment]
Very sorry for what I have said! Hope it did not affect to you! thanks for your reply! I made  a issue on it!------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;&quot;Xiao&amp;nbsp;Xi&amp;nbsp;LIU&quot;<notifications@github.com&gt;
发送时间:&nbsp;2020年11月5日(星期四) 下午5:16
收件人:&nbsp;&quot;hyperjumptech/grule-rule-engine&quot;<grule-rule-engine@noreply.github.com&gt;;
抄送:&nbsp;&quot;hellocowboy&quot;<956517404@qq.com&gt;;&quot;Comment&quot;<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [hyperjumptech/grule-rule-engine] Support logic NOT operator (#94)

**Comment by newm4n** [comment]
Made a pull request for this #135

**Comment by newm4n** [comment]
Fixed in new release v1.7.0
